amclient/amclient.py

```diff
@@ -607,12 +607,50 @@ def extract_file(self):
                         file_.write(chunk)
             return response.headers
 
+    def list_location_purposes(self):
+        """List valid location purposes in the Storage Service."""
+        return {
+            "AR": "AIP_RECOVERY",
+            "AS": "AIP_STORAGE",
+            "CP": "CURRENTLY_PROCESSING",
+            "DS": "DIP_STORAGE",
+            "SD": "SWORD_DEPOSIT",
+            "SS": "STORAGE_SERVICE_INTERNAL",
+            "BL": "BACKLOG",
+            "TS": "TRANSFER_SOURCE",
+            "RP": "REPLICATOR",
+        }
+
+    def create_location(self):
+        """Create a new location in the Storage Service."""
+        if not self.location_purpose.upper() in self.list_location_purposes():
+            return {
+                "error": "location purpose not permitted",
+                "valid_purposes": self.list_location_purposes(),
+            }
+        url = "{0}/api/v2/location/".format(self.ss_url)
+        params = {
+            "description": self.location_description
+            if self.location_description
+            else "",
+            "pipeline": ["/api/v2/pipeline/{}/".format(self.pipeline_uuid)],
```

Comment:
Just curious about the list use here: I noticed that `Location.pipeline` is a `ManyToManyField`. Do we support locations associated to multiple pipelines in reality?

Reply:
I believe so. I don't think it's as cool as being able to point one pipeline at multiple different locations, especially multiple AIP locations for different types of material, but I could see it might be useful if you're processing different materials via different pipelines but want to do the inverse and have it end up all in the same place. It all comes down to intellectual control though I guess. 

WRT this PR I'll see what I can do to make it work with the client. it depends on complexity vs. time though, so I might follow up with an issue.